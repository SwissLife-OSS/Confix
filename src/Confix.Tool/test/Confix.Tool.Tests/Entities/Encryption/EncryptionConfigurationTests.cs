using System.Text.Json.Nodes;
using Confix.Tool.Abstractions;
using ConfiX.Entities.Component.Configuration;

namespace Confix.Tool.Tests.Entities.Encryption;

public class EncryptionConfigurationTests : ParserTestBase
{
    [Fact]
    public void Parse_Should_BeValid()
    {
        ExpectValid(
            """
            {
                "provider": {
                    "type": "test"
                }
            }
            """);
    }

    [Fact]
    public void Parse_Should_BeValid_WhenEmpty()
    {
        ExpectValid(""" { } """);
    }

    [Fact]
    public void Parse_Should_BeInvalid_When_ProviderIsNotAnObject()
    {
        ExpectInvalid(
            """
            {
                "provider": "test"
            }
            """);
    }


    public override object Parse(JsonNode json)
       => EncryptionConfiguration.Parse(json);
}