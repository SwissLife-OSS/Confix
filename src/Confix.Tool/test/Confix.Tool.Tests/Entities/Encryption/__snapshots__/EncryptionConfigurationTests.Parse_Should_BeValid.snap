{
  "Provider": {
    "Type": "test",
    "EnvironmentOverrides": null,
    "Values": {
      "type": "test"
    }
  }
}
